webapp/components/features/FollowList/FollowList.vue

```diff
@@ -0,0 +1,165 @@
+          :key="connection.id"
+          class="connections__item"
+        >
+          <user-teaser :user="connection" />
+        </li>
+      </ul>
+      <base-button
+        v-if="hasMore"
+        :loading="loading"
+        class="spacer-x-small"
+        size="small"
+        @click="$emit('fetchAllConnections', type)"
+      >
+        {{
+          $t('profile.network.andMore', {
+            number: allConnectionsCount - connections.length,
+          })
+        }}
+      </base-button>
+      <ds-input
+        v-if="!hasMore"
+        v-focus="true"
+        :name="`${type}Filter`"
+        :placeholder="filter"
+        class="spacer-x-small"
+        icon="filter"
+        size="small"
+        @input.native="setFilter"
+      />
+    </template>
+    <p v-else class="nobody-message">{{ userName }} {{ $t(`profile.network.${type}Nobody`) }}</p>
+  </base-card>
+</template>
+
+<script>
+import { escape } from 'xregexp/xregexp-all.js'
+import UserTeaser from '~/components/UserTeaser/UserTeaser'
+
+export default {
+  name: 'FollowerList',
+  components: {
+    UserTeaser,
+  },
+  props: {
+    user: { type: Object, default: null },
+    type: { type: String, default: 'following' },
+    loading: { type: Boolean, default: false },
+  },
+  data() {
+    return {
+      filter: null,
+    }
+  },
+  computed: {
+    userName() {
+      const { name } = this.user || {}
+      return name || this.$t('profile.userAnonym')
+    },
+    allConnectionsCount() {
+      return this.user[`${this.type}Count`]
+    },
+    connections() {
+      return this.user[this.type]
+    },
+    hasMore() {
+      return this.allConnectionsCount > this.connections.length
+    },
+    connectionsClass() {
+      return `connections${this.hasMore ? '' : ' --overflow'}`
+    },
+    filteredConnections() {
+      if (!this.filter) {
+        return this.connections
+      }
+
+      // @example
+      //  this.filter = 'foo';
+      //  fuzzyExpression = /([^f]*f)([^o]*o)([^o]*o)/i
+      const fuzzyExpression = new RegExp(
+        `${this.filter.split('').reduce((expr, c) => `${expr}([^${escape(c)}]*${escape(c)})`, '')}`,
+        'i',
+      )
+
+      const fuzzyScores = this.connections
+        .map((user) => {
+          const match = user.name.match(fuzzyExpression)
+
+          if (!match) {
+            return false
+          }
+
+          let score = 1
+          for (let i = 1; i <= this.filter.length; i++) {
+            score *= match[i].length
+          }
+
+          return {
+            user,
+            score,
+          }
+        })
+        .filter(Boolean)
+        .sort((a, b) => a.score - b.score)
+
+      return fuzzyScores.map((score) => score.user)
+    },
+  },
+  methods: {
+    setFilter(evt) {
+      this.filter = evt.target.value
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+.follow-list {
+  display: flex;
+  flex-direction: column;
+  position: relative;
+  width: auto;
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-26T18:21:14Z" title="Thursday, March 26th 2020, 7:21:14 pm +01:00">Mar 26, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,157 @@
+<template>
+  <base-card class="hc-connections">
+    <ds-space v-if="this.connections && this.connections.length" margin="x-small">
+      <ds-text tag="h5" color="soft">
+        {{ userName | truncate(15) }} {{ $t(`profile.network.${type}`) }}
+      </ds-text>
+    </ds-space>
+    <template v-else>
+      <p class="no-connections">{{ userName }} {{ $t(`profile.network.${type}Nobody`) }}</p>
+    </template>
+    <template v-if="this.connections && this.connections.length <= 7">
+      <ds-space v-for="follow in uniq(this.connections)" :key="follow.id" margin="x-small">
+        <!-- TODO: find better solution for rendering errors -->
+        <client-only>
+          <user-teaser :user="follow" />
+        </client-only>
+      </ds-space>
+      <ds-space v-if="this.allConnectionsCount - this.connections.length" margin="small">
+        <base-button
+          @click="fetchConnections"
+          :loading="this.isLoading"
+          size="small"
+          color="softer"
+        >
+          {{
+            $t('profile.network.andMore', {
+              number: this.allConnectionsCount - this.connections.length,
+            })
+          }}
+        </base-button>
+      </ds-space>
+    </template>
+    <template v-else-if="this.connections.length > 7">
+      <div class="overflow-container">
+        <ds-space
+          v-for="follow in uniq(this.filteredConnections)"
+          :key="follow.id"
+          margin="x-small"
+        >
+          <client-only>
+            <user-teaser :user="follow" />
+          </client-only>
+        </ds-space>
+      </div>
+      <ds-space margin="x-small">
+        <ds-input
+          @input.native="setFilter"
+          :placeholder="filter"
+          v-focus="true"
+          size="small"
+          icon="filter"
+        />
+      </ds-space>
+    </template>
+  </base-card>
+</template>
+
+<script>
+import uniqBy from 'lodash/uniqBy'
+import UserTeaser from '~/components/UserTeaser/UserTeaser'
+import { followedByQuery, followingQuery } from '~/graphql/User'
+
+export default {
+  name: 'FollowerList',
+  components: {
+    UserTeaser,
+  },
+  props: {
+    user: { type: Object, default: null },
+    type: { type: String, default: 'following' },
+  },
+  data() {
+    return {
+      additionalConnections: [],
+      filter: null,
+      isLoading: false,
+      queries: {
+        followedBy: followedByQuery,
+        following: followingQuery,
+      },
+    }
+  },
+  computed: {
+    userName() {
+      const { name } = this.user || {}
+      return name || this.$t('profile.userAnonym')
+    },
+    connections() {
+      return [...this.user[this.type], ...this.additionalConnections]
+    },
+    filteredConnections() {
+      if (!this.filter) {
+        return this.connections
+      }
+
+      const fuzzyExpression = new RegExp(
+        `${this.filter.split('').reduce((part, c) => `${part}[^${c}]*${c}`)}`,
+        'i',
+      )
+
+      const fuzzyScores = this.connections
+        .map((user) => {
+          const match = user.name.match(fuzzyExpression)
+
+          return {
+            user,
+            score: match ? match[0].length * (match.index + 1) : -1,
+          }
+        })
+        .filter((score) => score.score !== -1)
+        .sort((a, b) => a.score - b.score)
+
+      return fuzzyScores.map((score) => score.user)
+    },
+    allConnectionsCount() {
+      return this.user[`${this.type}Count`]
+    },
+  },
+  methods: {
+    uniq(items, field = 'id') {
+      return uniqBy(items, field)
+    },
+    async fetchConnections() {
+      this.$set(this, 'isLoading', true)
+      const { data } = await this.$apollo.query({
+        query: this.queries[this.type],
+        variables: { id: this.user.id },
+        // neither result nor update are being called when defined here (?)
+      })
+      this.additionalConnections = data.User[0][this.type]
```


I think we could use the same query for `User` and update the apollo cache by using `updateQuery`
there are some examples of this, that have some differences, but you could check them out, or we could look together. For example, `/pages/index.vue`
Also, see https://vue-apollo.netlify.com/api/smart-query.html#fetchmore
It would be nice to take advantage of some of the power of `smart` queries, reactive variables, and apollo caching to avoid adding extra queries.

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-01T09:30:20Z" title="Wednesday, April 1st 2020, 11:30:20 am +02:00">Apr 1, 2020</time>_

---


done in e6361c9